iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/TokenRenewalTests.java

```diff
@@ -253,17 +254,16 @@ else if (protocol == AMQPS || protocol == AMQPS_WS)
         return clients;
     }
 
-    private MultiplexingClient createMultiplexedClientToTest(IotHubClientProtocol protocol, List<DeviceClient> createdClients) throws IotHubException, IOException, URISyntaxException, MultiplexingClientException, InterruptedException {
-        String hostName = createdClients.get(0).getConfig().getIotHubHostname();
-        MultiplexingClient multiplexingClient = new MultiplexingClient(hostName, protocol);
+    private MultiplexingClient createMultiplexedClientToTest(IotHubClientProtocol protocol, List<DeviceClient> clientsToCreate, String hostname) throws IotHubException, IOException, URISyntaxException, MultiplexingClientException, InterruptedException {
+        MultiplexingClient multiplexingClient = new MultiplexingClient(hostname, protocol);
```

Comment:
I wonder why hostname gets null in case of device client and not for internal client list?

Reply:
"clientsToCreate" is by design empty when this method is called. The method's job is to add clients to the list, not to read from the list. Because of that, it didn't make sense to try to get the hostname from the empty list